plumber/api/importer.py

```diff
@@ -0,0 +1,573 @@
+            return
+
+        if context is None:
+            import bpy
+
+            context = bpy.context
+
+        try:
+            # Use the new Rust API importer for better performance
+            import plumber
+
+            callbacks = _create_asset_callbacks(context)
+            threads = _get_threads_suggestion(context)
+
+            # Create API importer with material settings
+            api_importer = plumber.ApiImporter(
+                self._file_system._fs,
+                callbacks,
+                threads,
+                **self._material_settings,
+            )
+
+            # Add all jobs to the API importer
+            for job in self._jobs:
+                if job.asset_type == AssetType.VMF:
+                    # VMF jobs don't pass material options since they're already in the importer
+                    non_material_options = {
+                        k: v
+                        for k, v in job.options.items()
+                        if k not in self._material_settings
+                    }
+                    api_importer.add_vmf_job(
+                        job.path, job.from_game, **non_material_options
+                    )
+                elif job.asset_type == AssetType.MDL:
+                    # MDL jobs don't pass material options since they're already in the importer
+                    non_material_options = {
+                        k: v
+                        for k, v in job.options.items()
+                        if k not in self._material_settings
+                    }
+                    api_importer.add_mdl_job(
+                        job.path, job.from_game, **non_material_options
+                    )
+                elif job.asset_type == AssetType.VMT:
+                    api_importer.add_vmt_job(job.path, job.from_game)
+                elif job.asset_type == AssetType.VTF:
+                    api_importer.add_vtf_job(job.path, job.from_game)
+
+            # Execute all jobs
+            api_importer.execute_jobs()
+
+            # Clear jobs after execution
+            self._jobs.clear()
+
+        except Exception as e:
+            raise AssetImportError(f"Parallel import execution failed: {e}") from e
+
+    def clear(self) -> "ParallelImportBuilder":
+        """Clear all import jobs."""
+        self._jobs.clear()
+        return self
+
+    @property
+    def job_count(self) -> int:
+        """Get the number of queued import jobs."""
+        return len(self._jobs)
+
+
+def _create_asset_callbacks(context, **options) -> Any:
+    from ..asset import AssetCallbacks
+
+    # Extract collection options - default to main collection if not specified
+    main_collection = options.get("main_collection")
+    if main_collection is None:
+        main_collection = context.scene.collection
+
+    brush_collection = options.get("brush_collection")
+    if brush_collection is None:
+        brush_collection = context.scene.collection
```

Comment:
All the other collections here should default to main collection for cases where only main collection is specified